command/command.go

```diff
@@ -175,70 +177,84 @@ func (w *backgroundCmd) Start() error {
 // KeepAlive will create a goroutine to send regular SIGCONT signal to associated command
 // a single goroutine will be launched no matter how many calls are done to KeepAlive
 func (w *backgroundCmd) KeepAlive() {
+	w.Lock()
+	defer w.Unlock()
+
 	if w.DryRun() {
 		return
 	}
 
-	w.Lock()
-	defer w.Unlock()
-
 	if w.ticker != nil {
 		return
 	}
 
 	w.ticker = time.NewTicker(cmdKeepAliveTickDuration)
 
+	w.chQuit = make(chan int)
+
 	w.log.Debug("monitoring sending SIGCONT signal to process every 1s")
 
 	go func() {
 		for {
-			if w.ticker == nil {
+			if err := w.sendSIGCONTSignal(); err != nil {
 				return
 			}
+		}
+	}()
+}
 
-			<-w.safeTicker().C
+func (w *backgroundCmd) sendSIGCONTSignal() error {
+	w.Lock()
+	defer w.Unlock()
 
-			proc, err := w.processManager.Find(w.pid)
-			if err != nil {
-				w.log.Errorw("an error occurred when trying to Find process, stopping to monitor background process, ticker removed", "error", err)
+	if w.ticker == nil {
+		return fmt.Errorf("ticker is nil")
+	}
 
-				w.resetTicker()
+	select {
+	case <-w.chQuit:
```

Comment:
I don't understand this select block, won't we expect that anytime `resetTicker` is called, we end up returning an error here? I must be misunderstanding the control flow

Reply:
You right, in fact it's not an error. I have changed the code to be more explicit 👍